src/common/snippets/src/lowered/pass/compute_buffer_allocation_size.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/compute_buffer_allocation_size.hpp"
+
+#include "snippets/op/buffer.hpp"
+#include "snippets/utils.hpp"
+#include "snippets/itt.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+namespace {
+std::vector<size_t> get_parent_inner_loops(const std::vector<size_t>& parent_loops, const std::vector<size_t>& current_loops) {
+    const auto common_rank = std::min(parent_loops.size(), current_loops.size());
+    size_t i = 0;
+    while (i < common_rank && parent_loops[i] == current_loops[i])
+        ++i;
+    return std::vector<size_t>(parent_loops.cbegin() + i, parent_loops.cend());
+}
+}  // namespace
+
+// Ticket: 113744
+// TODO: This logic covers only several specific cases so it should be generalized.
+size_t ComputeBufferAllocationSize::get_allocation_size(const LoopManagerPtr& loop_manager, const ExpressionPtr& buffer_expr, size_t allocation_rank) {
+    const auto& parent_port = buffer_expr->get_input_port_connector(0)->get_source();
+    const auto& parent_loop_ids = get_parent_inner_loops(parent_port.get_expr()->get_loop_ids(), buffer_expr->get_loop_ids());
+    const auto planar_shape = utils::get_preordered_vdims(parent_port);
+
+    const size_t rank = allocation_rank >= 0 ? std::min(static_cast<size_t>(allocation_rank), planar_shape.size())
+                                             : planar_shape.size();
+
+    const auto& subtensor =  parent_port.get_descriptor_ptr()->get_subtensor();
+
+    size_t allocation_size = 1;
+    std::set<size_t> processed_dim_idxs;
+    for (const auto& parent_loop : parent_loop_ids) {
+        const auto loop_info = loop_manager->get_loop_info(parent_loop);
+        const auto& output_ports = loop_info->get_output_ports();
+        auto it = std::find_if(output_ports.begin(), output_ports.end(), [&parent_port](const LoopPort& port) { return *port.expr_port == parent_port; });
+        OPENVINO_ASSERT(it != output_ports.end(), "compute_allocation_shape: output port of parent loop can not be found");
+        const auto& loop_port = *it;
+        const auto& dim_idx = loop_port.dim_idx;
+        if (loop_port.is_incremented && dim_idx < rank) {
+            if (const auto& unified_loop_info = ov::as_type_ptr<UnifiedLoopInfo>(loop_info))
+                allocation_size = utils::dynamic_safe_mul(allocation_size, unified_loop_info->get_work_amount());
+            else if (const auto& expanded_loop_info = ov::as_type_ptr<ExpandedLoopInfo>(loop_info))
+                allocation_size = utils::dynamic_safe_mul(allocation_size, expanded_loop_info->get_unified_loop_info()->get_work_amount());
```

Comment:
No, I'm not sure in this idea because if `ExpandedLoopInfo->get_work_amount()` returns `get_unified_loop_info()->get_work_amount()`, how can we get `work_amount` of `ExpandedLoopInfo`? I think that each `LoopInfo` should return the exact `work_amount` value of the corresponding loop (not of whole loop)

Reply:
Oh, my bad, sorry. Obviously, we can't change the logic per my suggestion